reconciler/events_test.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Veroute.on 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciler
+
+import (
+	"errors"
+	"io"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var exampleStatusFailed = "ExampleStatusFailed"
+
+func TestNil_Is(t *testing.T) {
+	var err error
+	if EventIs(err, NewEvent(corev1.EventTypeWarning, exampleStatusFailed, "")) {
+		t.Error("not a ReconcilerEvent")
```

Comment:
I find the failure message confusing. Did you mean "expected nil error not to be a ReconcilerError"? Future devs will thank you for the clear logs :)

Reply:
Sorry for lazy error strings... Updated.